server/src/main/java/org/opensearch/index/codec/CodecAliases.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec;
+
+import org.apache.lucene.codecs.Codec;
+
+import java.util.Set;
+
+/**
+ * This {@link CodecAliases} to provide aliases for the {@link Codec}.
+ *
+ * @opensearch.internal
+ */
+public interface CodecAliases {
+
+    /**
+     * Retrieves a set of aliases for an codec.
+     *
+     * @return A non-null set of alias strings. If no aliases are available, an empty set should be returned.
+     */
+    Set<String> aliases();
```

Comment:
Can it have default implementation returning empty set, so that implementing classes need not add redundant code if there are no aliases.

```suggestion
    default Set<String> aliases() {
        return Collections.EMPTY_SET;
    }
```